### DISCUSSION TITLE
Reactive kafka consumer

### DISCUSSION THREAD

**Discussion by vasantteja** [spring-projects/spring-kafka#2351] [Q&A] [answered]
I was thinking of building a reactive kafka consumer and producer. I came across this [https://stackoverflow.com/questions/65627958/how-do-you-use-a-kafkalistener-annotated-method-with-a-flux-or-mono](url). Is there way we can create a reactive consumer and producer with spring-kafka?

**Comment by garyrussell** [comment, marked as answer]
No; we started investigating some time ago but came to the conclusion that it wouldn't add much benefit over using the `KafkaReceiver` from `reactor-kafka`.

spring-kafka has a couple of lightweight templates on top of reactor-kafka but, again, they don't really add much value.

**Reply by vasantteja** [reply]
Thanks @garyrussell  this was really helpful.

**Reply by ahmedalnuaimi** [reply]
Can you point me to the discussion where you reached that conclusion? I need that to convince my company that it doesn't have the same benefit of Webflux migration.

**Reply by garyrussell** [reply]
Let me turn that around - exactly what functionality do you need that you can't achieve with reactor-kafka?

Spring would not be involved with the messaging at runtime, only to set up the topology, so the framework can't help with error handling, for example.

**Reply by ahmedalnuaimi** [reply]
I want to integrate it using Autoconfiguration just the way WebFlux is integrated.

**Reply by garyrussell** [reply]
Right, but that is spring-boot, it does not mean more integration in this project. 

There is an open PR there; but it hasn't had much movement.

**Reply by ahmedalnuaimi** [reply]
Ah OK so this PR is not related to the Spring Boot effort, right?

**Reply by garyrussell** [reply]
No; this was about adding something like 

```java
@KafkaListener(...)
void listen(Flux<Foo> flux) {
   ...
}
```

Which doesn't make much sense because Spring  would only be involved in setting up the topology; it would not be involved with the runtime record reception/processing.
